src/test/regress/sql/qp_dropped_cols.sql

```diff
@@ -8687,3 +8687,114 @@ DELETE FROM dist_key_dropped_pt WHERE b=6;
 -- the tables, or the pg_upgrade test fails.
 set client_min_messages='warning';
 drop schema qp_dropped_cols cascade;
+
+-- Test modifying DML on leaf partition when parent has dropped columns and
+-- the partition has not. Ensure that DML commands pass without execution
+-- errors and produce valid results.
+RESET search_path;
+-- start_ignore
+DROP TABLE IF EXISTS t_part_dropped;
+-- end_ignore
+CREATE TABLE t_part_dropped (c1 int, c2 int, c3 int, c4 int) DISTRIBUTED BY (c1)
+PARTITION BY LIST (c3) (PARTITION p0 VALUES (0));
+
+ALTER TABLE t_part_dropped DROP c2;
+ALTER TABLE t_part_dropped ADD PARTITION p2 VALUES (2);
+
+-- Partition selection should go smoothly when inserting into leaf
+-- partition with different attribute structure.
+INSERT INTO t_part_dropped VALUES (1, 2, 4);
+INSERT INTO t_part_dropped_1_prt_p2 VALUES (1, 2, 4);
+INSERT INTO t_part_dropped_1_prt_p2 VALUES (1, 2, 0);
+
+-- Ensure that split update on leaf and root partitions does not
+-- throw partition selection error in both planners.
+UPDATE t_part_dropped_1_prt_p2 SET c1 = 2;
+UPDATE t_part_dropped SET c1 = 3;
+
+INSERT INTO t_part_dropped VALUES (1, 2, 0);
+
+-- Ensure that split update on leaf partition does not throw constraint error
+-- (legacy planner does not choose the wrong partition at insert) and does not
+-- throw partition selection error (ORCA).
+UPDATE t_part_dropped_1_prt_p2 SET c1 = 2 WHERE c4 = 0;
```

Comment:
Is this insert is a preparation for the below test? Maybe then move it under the comment so that it is clear what it refers to? same on the line 8757

Reply:
Done